### PR TITLE
Make builtins.sum use torch.sym_add when applicable

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136530
* #136429
* #136533

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -1002,6 +1004,35 @@ def call_function(
     ) -> "VariableTracker":
         from torch._dynamo.variables.builder import SourcelessBuilder
 
+        # Special case for sum on tuple of floats and ints
```
Comment (isuruf) [inline torch/_dynamo/variables/functions.py:1007]: No floats right?

Comment (ezyang) [reply torch/_dynamo/variables/functions.py:1007]: oopsy yep

File: torch/_dynamo/variables/functions.py [hunk 2]
```diff
@@ -1002,6 +1004,35 @@ def call_function(
...
+        if (
+            self.fn is builtins.sum
+            and len(args) == 1
+            and not kwargs
```
Comment (XuehaiPan) [inline torch/_dynamo/variables/functions.py:1011]: `sum` supports a keyword-only argument `start`:

```python
sum(iterable: Iterable[T], /, start: U = 0) -> R
```

Comment (ezyang) [reply torch/_dynamo/variables/functions.py:1022]: I'm laaaazy lol

**Comment by ezyang** [comment]
Gonna fold this into prev PR

**Pull request closed without merging**